src/views/ac-forms/ac-forms-product/ac-forms-product.js

```diff
@@ -315,16 +315,26 @@ const AcFormsProductInner = ({
         : [];
       if (rcIds.length === 0) return;
 
+      // ✅ FIX: Normalize referentieComponenten IDs the same way as the stage component
+      const normalizeReferentieComponentenId = (refId) => {
+        if (typeof refId === 'object' && refId !== null) {
+          return String(refId.id || refId.value || refId.naam || refId);
+        }
+        return String(refId);
+      };
```

Comment:
a early return for null would be nice here.
Since if `return String(refId);` runs and the value is null it returns `'null'`

```suggestion
      const normalizeReferentieComponentenId = (refId) => {
        if (refId === null || refId === undefined) return refId
        if (typeof refId === 'object' && refId !== null) {
          return String(refId.id || refId.value || refId.naam || refId);
        }
        return String(refId);
      };
```